ccdaservice/utils/demographics/populate-demographics.js

```diff
@@ -0,0 +1,169 @@
+'use strict';
+
+const { fetchPreviousAddresses } = require('./previous-addresses');
+const { fDate } = require('../date/date');
+const {
+    NOT_INFORMED,
+    NULL_FLAVOR,
+    DECLINED_TO_SPECIFY,
+} = require('../constants');
+
+function getGuardianNames(display_name) {
+    const parts = display_name.split(' ');
+    return parts.length === 3
+        ? [{ first: parts[0], last: parts[2] }]
+        : parts.length === 2
+        ? [{ first: parts[0], last: parts[1] }]
+        : [{ first: NOT_INFORMED, last: NOT_INFORMED }];
+}
+
+function getGuardianInfo(guardian) {
+    return [
+        {
+            relation: guardian.relation,
+            addresses: [
+                {
+                    street_lines: [guardian.address],
+                    city: guardian.city,
+                    state: guardian.state,
+                    zip: guardian.postalCode,
+                    country: guardian.country || 'US',
+                    use: 'primary home',
+                },
+            ],
+            names: getGuardianNames(guardian.display_name),
+            phone: [
+                {
+                    number: guardian.telecom,
+                    type: 'primary home',
+                },
+            ],
+        },
+    ];
+}
+
+function setNullFlavorIfUnspecifiedOrEmpty(patient, property) {
+    if (patient[property] === DECLINED_TO_SPECIFY || patient[property] === '') {
+        patient[property] = NULL_FLAVOR;
+    }
+}
+
+function getLanguageCode(patient) {
+    return patient.language === 'English'
+        ? 'en-US'
+        : patient.language === 'Spanish'
+            ? 'sp-US'
+            : 'en-US';
+}
+
+function populateDemographics(patient, guardian, documentData) {
+    const oidFacility =
+        documentData.encounter_provider.facility_oid ||
+        '2.16.840.1.113883.19.5.99999.1';
+    const npiFacility =
+        documentData.encounter_provider.facility_npi || NOT_INFORMED;
```

Comment:
I'm not sure this change is correct and would need to be tested / looked at.  You took the original line from 3228 in generateUnstructured which sets npiFacility to be: 
```
npiFacility = all.encounter_provider.facility_npi || "NI";
```
However in generateCcda on line 2754 on npiFacility does not have an option for "NI":
```
npiFacility = all.encounter_provider.facility_npi;  
```
@sjpadgett does the spec allow "NI" for a structured document?

Reply:
Good point, I'm not sure if the abscence of a fallback value in line 2754 is intentional or not so it'll be good to clarify that. In any case, if it's necessary to prevent "NI" in structured documents, it should be easy to accommodate for that in the refactored version

Reply:
Leave any attributes or replace values as they may have significance to builder. Unless a default is set.

Reply:
Latest changes keep the fallback and switch between unstructured and structured.  I consider this now resolved.